File: Exemplos SmartPOS/App_IntentDigitalHUB_Java/Elgin_IntentDigitalHUB_SmartPOS_Java-font/app/src/main/java/com/elgin/java_intentdigitalhub_smartpos/IntentDigitalHubService/ELGINPAY/Commands/IniciaVendaCredito.java
package com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.ELGINPAY.Commands;

import androidx.annotation.NonNull;

import com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.ELGINPAY.ElginPayCommand;
import com.google.gson.JsonObject;

public class IniciaVendaCredito extends ElginPayCommand {
    final private String valorTotal;
    final private int tipoFinanciamento;
    final private int numeroParcelas;

    public IniciaVendaCredito(@NonNull String valorTotal, int tipoFinanciamento, int numeroParcelas) {
        super("iniciaVendaCredito");
        this.valorTotal = valorTotal;
        this.tipoFinanciamento = tipoFinanciamento;
        this.numeroParcelas = numeroParcelas;
    }

    @Override
    protected JsonObject functionParametersJson() {
        JsonObject functionParametersJson = new JsonObject();

        functionParametersJson.addProperty("valorTotal", this.valorTotal);
        functionParametersJson.addProperty("tipoFinanciamento", this.tipoFinanciamento);
        functionParametersJson.addProperty("numeroParcelas", this.numeroParcelas);

        return functionParametersJson;
    }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_SmartPOS_Kotlin-font/app/src/main/java/com/elgin/kotlin_intent_digital_hub_smartpos/IntentDigitalHubService/ELGINPAY/Commands/IniciaVendaCredito.kt
package com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.ELGINPAY.Commands

import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.ELGINPAY.ElginPayCommand
import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.IntentDigitalHubModule
import com.google.gson.JsonObject

class IniciaVendaCredito(private val valorTotal: String, private val tipoFinanciamento: Int, private val numeroParcelas: Int ) : ElginPayCommand(functionName ="iniciaVendaCredito") {

    override fun functionParametersJson(): JsonObject {
        val functionParametersJson = JsonObject()
        functionParametersJson.addProperty("valorTotal", this.valorTotal)
        functionParametersJson.addProperty("tipoFinanciamento", this.tipoFinanciamento)
        functionParametersJson.addProperty("numeroParcelas", this.numeroParcelas)
        return functionParametersJson
    }
}